### DISCUSSION TITLE
Strange attributes of HGTConv

### DISCUSSION THREAD

**Discussion by yamaceay** [pyg-team/pytorch_geometric#7180] [General]
Hey, I recently noticed that two strange variables are defined in `HGTConv.__init__()` (torch_geometric/nn/conv/hgt_conv.py), causing low-level errors:

```python
class HGTConv(...):
  def __init__(self, ...):
    ...
    # metadata looks like: [node_types, edge_types]
    # node_types is a list of all node types
    # edge_types is a list of tuples for each meta relation type (src_node_type, edge_type, dest_node_type)

    self.dst_node_types = list(set(metadata[1][1]))
    self.src_types = [edge_type[0] for edge_type in self.edge_types]
```

`self.dst_node_types`: It is assigned the unique values of second meta relation type. 

For example:
```python
metadata = [["X", "Y", "Z"], [["X", "a", "Z"], ["X", "b", "X"]]]
self.dst_node_types = list(set(metadata[1][1]))
assert(sorted(self.dst_node_types) == ["b", "X"]) # returns: True
```
The variable is used at one line, which causes an error at my code (torch_geometric/nn/conv/hgt_conv.py:212-215): 

```python
  def forward(self, ...):
    ...
    for node_type, out in out_dict.items():
        if node_type not in self.dst_node_types:
            out_dict[node_type] = None
            continue
```

`self.src_types`: It is assigned a non-unique list of all src nodes occurring at each edge type. The variable is unused in HGTConv as well as in its parent-classes. 

Is there any possibility that it is a bug which hasn't been fixed yet? Or can you explain why and how these variables are used?
Thank you in advance!

**Comment by rusty1s** [comment]
Thanks for letting us know. This was mistakenly introduced by https://github.com/pyg-team/pytorch_geometric/pull/7117, and I fixed it in https://github.com/pyg-team/pytorch_geometric/pull/7183. Let me know if you have any doubts.
